### DISCUSSION TITLE
如何使标题页“学院（系）”内容自动换行？

### DISCUSSION THREAD

**Discussion by LongyuanZhang** [sjtug/SJTUThesis#610] [提问] [answered]
如题，标题页的学院全称过长，导致跑出了页面。有没有办法让内容自动换行？我试过修改\SJTUThesis-master\texmf\tex\latex\sjtuthesis的sjtuthesis-undergraduate.ltx，发现标题页这块内容是通过tabular实现的。但不管是用\begin{tabular}{m{***cm}}还是用multirow宏包，效果都不行。求助各位，谢谢！

**Comment by skyzh** [comment]
可以尝试用 `\\` 换行？

**Reply by LongyuanZhang** [reply]
感谢回复！尝试过不管是在sjtusetup.tex还是在sjtuthesis-undergraduate.ltx中用\\换行，都不行，当然也可能是我用法不对。现在勉强的解决办法是再加加一个tabular然后用\quad控制内容长度（听上去有点蠢

**Comment by skyzh** [comment, marked as answer]
首先在 `main.tex` 中加入 `\usepackage{makecell}`

而后在 `sjtusetup.tex` 中如此设置
```
department      = {\makecell{某个名字很长很长很长很长很长很长\\很长很长很长很长很长很长很长\\很长很长很长很长很长的系}},
```

即可

<img width="741" alt="image" src="https://user-images.githubusercontent.com/4198311/118150236-e81e2e80-b444-11eb-9d46-49a9e8349d9c.png">

**Reply by skyzh** [reply]
<del>但是似乎不能自动换行</del>

**Reply by LongyuanZhang** [reply]
效果挺好的！不过这样好像没办法两行文字都带有下划线，当然也是我过于带有word的排版思维了。谢谢！

**Comment by skyzh** [comment]
本科的模版可以直接通过这个方法设置：

```
department      = {\parbox[t]{280pt}{某个名字很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长很长的系}},
```

<img width="693" alt="image" src="https://user-images.githubusercontent.com/4198311/118152298-1bfa5380-b447-11eb-9b1c-fe3dd27425d7.png">
